src/logging.ts

```diff
@@ -99,21 +114,159 @@ export class Logger {
     const formattedMessage = `${prefix} ${message} ${argString}\n`;
 
     return new Promise<void>((resolve, reject) => {
-      const logWriteStream = createWriteStream(LOGFILE_PATH, { flags: "a" });
-      logWriteStream.once("error", (error) => {
-        console.error("Error writing to log file:", error);
-        logWriteStream.end();
-        reject(error);
-      });
-      logWriteStream.write(Buffer.from(formattedMessage), (error) => {
-        if (error) {
-          console.error("Error writing to log file:", error);
-          logWriteStream.end();
-          reject(error);
+      try {
+        const stream: RotatingFileStream = getLogFileStream();
+        if (stream.closed) {
+          resolve();
+          return;
         }
-        // waits for any remaining data to be written to the file before closing the stream
-        logWriteStream.end(() => resolve());
-      });
+        stream.write(formattedMessage, (error) => {
+          if (error) {
+            console.error("Error writing to log file:", error);
+            reject(error);
+          } else {
+            resolve();
+          }
+        });
+      } catch (error) {
+        console.error("Unexpected error during log file operations:", error);
+        resolve();
+      }
+    });
+  }
+}
+
+/**
+ * Default path to store downloadable log files for this extension instance.
+ *
+ * The main difference between this and the ExtensionContext.logUri (where LogOutputChannel lines
+ * are written) is this will include ALL log levels, not just the ones enabled in the output channel.
+ */
+export const LOGFILE_DIR = tmpdir();
+
+/** The name of the currently active log file, including time/index prefixing. */
+export let CURRENT_LOGFILE_NAME: string;
+
+/** Set of log file names that have already been created for this extension instance. */
+export const ROTATED_LOGFILE_NAMES: string[] = [];
+
+/** Max size of any log file written to disk.
+ * @see https://github.com/iccicci/rotating-file-stream?tab=readme-ov-file#size */
+const MAX_LOGFILE_SIZE = "10M"; // 10MB max file size
+
+/** Number of log files to keep.
+ * @see https://github.com/iccicci/rotating-file-stream?tab=readme-ov-file#maxfiles */
+export const MAX_LOGFILES = 3; // only keep 3 **rotated** log files at a time for this extension instance
```

Comment:
Refine comment to indicate "... : the current extension-free file, and then N-1 rotated extension files" or something like that please.

Saying "**rotated**" so boldly makes me think that ROTATED_LOGFILE_NAMES will only end up with the `.N`-extension rotated filenames.